src/goto-instrument/cover.cpp

```diff
@@ -1714,6 +1694,12 @@ bool instrument_cover_goals(
     }
   }
 
+
+  // cover entry point function only
+  std::string cover_include_pattern=cmdline.get_value("cover-include-pattern");
+  if(cmdline.isset("cover-function-only"))
+    cover_include_pattern=config.main;
```

Comment:
for `regex` we should escape `.`, else there could be a problem with correct matching, may also be required for other characters that can be part of a function name, maybe '<', '>' ?

Reply:
This appears to be a robust way of doing this: https://stackoverflow.com/a/40195721/958004